packages/studio-ui/src/utils/PageDataValidator.ts

```diff
@@ -28,12 +40,20 @@ export default class PageDataValidator {
    */
   validate(pageData: { pageName?: string; url?: string }): ValidationResult {
     const errorMessages: string[] = [];
-    if (pageData.pageName !== undefined)
-      errorMessages.push(...this.validatePageName(pageData.pageName));
-    if (pageData.url !== undefined)
-      errorMessages.push(
-        ...this.validateURLSlug(pageData.url, this.isEntityPage)
-      );
+    const { pageName, url } = pageData;
+    if (pageName !== undefined) {
+      errorMessages.push(...this.validatePageName(pageName));
```

Comment:
Can we move the page name uniqueness check into `validatePageName`? Then we have only one place where we are doing page name validation.

Reply:
that makes sense!